UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs

```diff
@@ -6,6 +6,6 @@ public class ChatAiOptions: OpenAiOptions
 {
 	public ChatAiOptions(IOptions<AppConfig> appConfig)
 	{
-		this.ApiKey = appConfig.Value.ApiKey ?? throw new InvalidOperationException("You must define an OpenAI API key in appsettings.json file.");
+		this.ApiKey = appConfig.Value.ApiKey!;
```

Comment:
I would not remove the null check here. If the app uses the mock, this `ChatAiOptions` should not be resolved from the DI so it won't throw any exception (we could even imagine to not register it at all when using mock).

Reply:
If we don't register `ChatAiOptions` when using mock, we can consider not registering `OpenAIService` as well. Then we will have something like:

```
.ConfigureServices(
	(context, services) => {

		var apiKey = context.Configuration.GetSection("AppConfig:ApiKey").Value;
		var useMockService = apiKey is null or { Length: 0 };

		if (useMockService)
		{
			services.AddSingleton<IChatService, MockChatService>();
		}
		else
		{
			services
				.AddSingleton<IChatCompletionService, OpenAIService>()
				.AddSingleton<IChatService, ChatService>()
				.AddSingleton<OpenAiOptions, ChatAiOptions>(); 
		}
		
		services.AddSingleton<BindableMainModel>();
	})
```

What do you think?

Reply:
Yes, if not used it's better that way 👍